### ISSUE TITLE
Initializing js/ts language features never completes

### ISSUE THREAD

**Issue by akdevservices** [microsoft/vscode#119268]
- VS Code Version: 1.54.3
- OS Version: Linux, Kubuntu 20.04

Steps to Reproduce:

1. Create Javascript file with sample content: `let x = 123;`
2. Set `"javascript.validate.enable": false` in settings.json.
3. Restart VS Code: `code --disable-extensions`
4. Pending action 'Initializing JS/TS language features' never completes. In my case it seems that it consumes some additional CPU time.

![1](https://user-images.githubusercontent.com/29972256/111667676-a0e93800-881d-11eb-9239-3500f43af0ca.png)

There is similar issue #68896 but it was closed without dealing with it.

Does this issue occur when all extensions are disabled?: Yes

**Comment by NewtMa** [comment]
same here

**Comment by dziku86** [comment]
Do you have `jsconfig.json` in your workspace folder?
Excluding unnecessary folders with JS files fixed this problem for me (at least for now).

```JSON
{
    "exclude": [
        "node_modules",
        ".history",
        ".vscode",
        "https",
        "polyfills"
    ]
}
```

**Comment by akdevservices** [comment]
@dziku86 
There's no local config files, just a single .js file in the current folder. But your workaround works, even an empty `jsconfig.json` does the job:
```
{
}
```
Though it is still a workaround and not a bug fix.

**Comment by kennu** [comment]
I've tried every possible trick, from adding exclude clauses in jsconfig.json / tsconfig.json files, to configuring Visual Studio Code Watcher Excludes and disabling JavaScript validation, but nothing seems to prevent VSCode from crawling my node_modules folders. It takes ages and everything is slow. I wish it would "just work".

**Comment by roblourens** [comment]
I can't repro the original issue in stable

**Comment by JacksonKearl** [comment]
![image](https://user-images.githubusercontent.com/8586769/116759008-24be4480-a9c6-11eb-9a09-8145f707dea4.png)

Still seeing this on Ubuntu. CPU pegged too.

**Comment by mjbvz** [comment]
That seems like a more serious issue but I can't reproduce it on Ubuntu. 
Based on your screenshot, I suspect that VS Code may be trying to process all the files in the subdirectories in your example. If that's the case, showing the loading indicator may be expected

A few things to check:

- If you trigger IntelliSense in that file, does it show `(loading...)`? 
- If you try this if you opened more scoped folder, do you still see this?

**Comment by JacksonKearl** [comment]
Opened a folder with just the single file and the CPU usage wasn't as bad, but the loading message still persisted for 10+ minutes.

**Comment by mjbvz** [comment]
Hm, can you please try collecting the TS Server log so I can investigate:

1. Set `"typescript.tsserver.log": "verbose",`
1. Restart VS Code and reproduce the problem
1. In VS Code, run the `TypeScript: Open TS Server log` command
1. This should open a large log file called `tsserver.log` 

Please share this log from when initializing is still running

**Comment by akdevservices** [comment]
@mjbvz 
[tsserver.log](https://github.com/microsoft/vscode/files/6431933/tsserver.log)

**Comment by akdevservices** [comment]
@mjbvz 
[tsserver.log](https://github.com/microsoft/vscode/files/6432081/tsserver.log)

**Comment by mjbvz** [comment]
@akdevservices What VS code version is this from?

**Comment by mjbvz** [comment]
Should be fixed more completely with f25ffb76edd1f0037da743ab4955ff7083309600

However you'd still see this if TS really is trying to load a gigantic project. @JacksonKearl Can you please test in the next insiders and share the server logs if you still hit this

**Comment by akdevservices** [comment]
@mjbvz latest stable 1.56.0

**Comment by akdevservices** [comment]
@mjbvz 
> Can you please test in the next insiders and share the server logs if you still hit this

seems to be fixed in code-insiders 1.57.0-1620624357

**Comment by connor4312** [comment]
I don't hit this, @JacksonKearl can you verify?
